examples/algorithm_progression/hdf5_sweep.py

```diff
@@ -0,0 +1,86 @@
+from pathlib import Path
+
+import h5py
+import numpy as np
+from algorithm_sweep import AlgorithmSweep
+
+import large_image
+
+
+class AlgorithmSweepHDF5(AlgorithmSweep):
+    def initialize_storage(self):
+        self.output_files = []
+        source = large_image.open(self.input_filename)
+        self.source_metadata = source.getMetadata()
+
+        self.filepath = Path(self.output_dir, 'results.h5')
+        self.iteration_dims = {
+            'y': self.source_metadata['sizeY'],
+            'x': self.source_metadata['sizeX'],
+            'c': 4,
+        }
+        self.param_dims = {k: len(v) for k, v in self.input_params.items()}
+        self.layout = h5py.VirtualLayout(
+            shape=tuple(dict(**self.param_dims, **self.iteration_dims).values()),
```

Comment:
We can make the h5py expandable and not need to know what size it will be before hand by starting with a shape of all 1s for the param_dims and something modest for the iteration_dims, and then specify a `maxshape=tuple([None] * (total number of dimensions))` and specify `chunks=True`.  If you start with an axis length of 1, h5py will chunk that axis in chunks of 1, which is super inefficient for the spatial dimensions.  We know the output image size, but we might not in other contexts, so we probably want to make the minimum x / y / bands be something like 1024 / 1024 / 4, and (if we make this a tile sink) track the actual used values.

I haven't used the VirtualLayout option, but `h5py.File(<path>, 'w').create_dataset('images', shape=..., maxshape=..., chunks=True)`.  You seemingly can't open h5py files for writing from multiple processes, so we might still need to collect each frame as a separate h5py file and then combine them at the end.